shared/libraries/shooter.h

```diff
@@ -25,34 +25,73 @@
 
 namespace control {
 
-typedef enum{
-	MOTOR_Left = 0,
-	MOTOR_Rright = 1,
-	MOTOR_Bullet = 2,
-} shooter_motor_t;
-
-typedef enum{
-	HERO = 0,
-	STANDARD = 1,
-	ENGINEER = 2,
-	AERIAL = 3,
-	SENTRY = 4,
-	DART = 5,
-	RADAR = 6
-} robot_type_t;
+/**
+ * @brief structure used when shooter instance is initialized
+ */
+typedef struct {
+  bool acc_using_can_motor;					 /* if accelerate motors are using CAN protocal  */
+  MotorCANBase* left_acc_can_motor;	 /* CAN motor instance of left accelerate motor  */
+  MotorCANBase* right_acc_can_motor; /* CAN motor instance of right accelerate motor */
+  MotorPWMBase* left_acc_pwm_motor;	 /* PWM motor instance of left accelerate motor  */
+  MotorPWMBase* right_acc_pwm_motor; /* PWM motor instance of right accelerate motor */
+  bool left_acc_motor_invert;				 /* if left accelrate motor is inverted					 */
+  bool right_acc_motor_invert;			 /* if right accelrate motor is inverted				 */
+  ServoMotor* load_servo;            /* servomotor instance of load motor            */
+  float acc_Kp;                      /* Kp of pid controlling accelerate motor speed */ 
+  float acc_Ki;                      /* Ki of pid controlling accelerate motor speed */
+  float acc_Kd;                      /* Kd of pid controlling accelerate motor speed */
+  float load_step_angle;             /* step size of loading motor, in [rad]         */
+} shooter_t;
 
+/**
+ * @brief wrapper class for shooter
+ */
 class Shooter {
 public:
-    Shooter(robot_type_t robot_type, const int* motor_id, const float* fire_pid, const float* load_pid);
-    void Fire(float speed);
-    void Load(float speed);
+  /**
+   * @brief constructor for Shooter instance
+   * 
+   * @param shooter structure that used to initialize gimbal, refer to type shooter_t
+   */
+  Shooter(shooter_t shooter);
+
+  /**
+   * @brief set the speed of accelerating motors
+   * 
+   * @param speed 
+   */
+  void SetAccSpeed(float speed);
+
+  /**
+   * @brief load the next bullet
+   * 
+   * @return int servomotor status, refer to type servo_status_t
+   */
+  int LoadNext();
+
+  /**
+   * @brief calculate the output of the motors under current configuration
+   * 
+   */
+  void CalcOutput();
```

Comment:
CalcOutput is a "standard" function I implemented in Gimbal, Shooter, and ServoMotor classes. All those functions have a common thing that no motor output is actually sent by any of these functions. They will perform some measurements and set the desired speed field of the motor. But no actual control command is sent through CAN lines until the next `control::MotorCANBase::TransmitOutput` is called. 

And also changing its name to `Fire` would not be appropriate since it not only handles the calculations of "fire motors" (i.e. accelerating motors), but also the load motor(s).